### PR TITLE
Just starting

### PR DESCRIPTION
**Pull request by ssilvonen**
Starting out, everything is still a mess. I don't understand how to use the existing code (https://www.npmjs.com/package/@natlibfi/marc-record-merge) together with this.

### CODE REVIEW

**Review by natlibfi-arlehiko — CHANGES_REQUESTED** [review]
I just updated the Jira issue but I'll summary it here too:
- This package's index.js exports all of it's own reducers and also a list of these reducers which can be fed to marc-record-merge to use all of Melinda's generic reducers.

The contents of index.js would look like this:
```js
import * as reducers from './reducers';
export * from './reducers';
export default reducers;
```

(This exports each reducer with their own name separately also, if some application wants only use certain reducers from the package).

Example of implementing a reducer: 010 (Specified [here](https://workgroups.helsinki.fi/x/3JwzCQ)):

`"010": { "action": "copy", "options": { "compareWithout": ["9"] } },`

This would otherwise work with copy reducer from `marc-record-merge` but subfield 9 needs to be excluded from comparison. Excluding subfields is non-Melinda specific feature which would be good to have in the copy reducer in marc-record-merge. This package (marc-marc-record-merge-reducers) would then export the copy reducer with parameters for field 010 with subfield 9 excluded from comparison.

File: src/reducers/field006.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import {MarcRecord} from '@natlibfi/marc-record';
+import createReducer from './field006';
+import fixturesFactory, {READERS} from '@natlibfi/fixura';
+import createDebugLogger from 'debug';
+
+MarcRecord.setValidationOptions({subfieldValues: false});
+
+describe('reducers/field006', () => {
+  const debug = createDebugLogger('@natlibfi/melinda-marc-record-merge-reducers');
+  const {expect} = chai;
```
Comment (natlibfi-jonollil) [inline src/reducers/field006.spec.js:13]: https://github.com/NatLibFi/melinda-commons-js/blob/ac0868f904c8f247046ff70128056bf40c25a6b0/src/subRecordPicker.spec.js#L29

File: src/reducers/field006.spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const fixturesPath = path.join(__dirname, '..', '..', 'test-fixtures', 'reducers', 'field006');
+
+  fs.readdirSync(fixturesPath).forEach(subDir => {
+    const {getFixture} = fixturesFactory({root: [fixturesPath, subDir], reader: READERS.JSON, failWhenNotFound: false});
+    it(subDir, () => {
+      const base = new MarcRecord(getFixture('base.json'));
+      const source = new MarcRecord(getFixture('source.json'));
+      const tagPattern = new RegExp(getFixture({components: ['tagPattern.txt'], reader: READERS.TEXT}), 'u');
```
Comment (natlibfi-jonollil) [inline src/reducers/field006.spec.js:21]: Since tag pattern stays as same for whole test and is one liner, it can be hard coded.

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

File: src/reducers/melindaCustomMergeFunctions.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Custom functions for Melinda merge
```
Comment (natlibfi-jonollil) [inline src/reducers/melindaCustomMergeFunctions.js:1]: File could be named as something more common, like 'utils.js' and there you could collect all functions what can be used in any reducer later on. (checkDropSubfields and/or compareSubValues)

**Pull request merged**